apps/server/src/controllers/groupControllers/helpers.ts

```diff
@@ -1,16 +1,170 @@
+//   groupTitle: string
+// }
+// export const createNewMember = (props: Props) => {
+//   return {
+//     id: v4(),
+//     userId: props.userId,
+//     groupId: props.groupId,
+//     groupTitle: props.groupTitle,
+//     amountContributed: 0,
+//     totalAmountWithdrawnByUser: 0,
+//     dateOfLastContribution: null,
+//   }
+// }
+
+export const getUserWithId = async (userId: string) => {
+  return await Users.findOne({
+    where: { id: userId },
+  })
+}
 
 interface Props {
   userId: string
   groupId: string
+  groupTitle: string
+  options?: Partial<{
+    isAdmin: boolean
+  }>
+}
+
+export const createNewMember = async (props: Props) => {
+  const user = await getUserWithId(props.userId)
+
+  if (user) {
+    let sequence
+    const MAX_SEQUENCE_VALUE = 1000000
+
+    if (props.options?.isAdmin) {
+      sequence = MAX_SEQUENCE_VALUE
+    } else {
+      // do something to get the next sequence value for a non-admin user
+    }
+
+    return {
+      id: v4(),
+      userId: props.userId,
+      groupId: props.groupId,
+      groupTitle: props.groupTitle,
+      amountContributed: 0,
+      totalAmountWithdrawnByUser: 0,
+      dateOfLastContribution: null,
+      isAdmin: props.options?.isAdmin || false,
+      sequence: sequence,
+    }
+  }
+  return null
+}
+
+async function determineNextBeneficiary(
+  userId: string
+): Promise<string | null> {
+  const nextMember = await GroupMembers.findOne({
+    where: {
+      userId,
+    },
+    order: [['sequence', 'ASC']],
+  })
+
+  return nextMember ? nextMember.userId : null
+}
+
+async function notifyBeneficiary(
+  userId: string | null,
+  user: Users
+): Promise<void> {
+  if (userId) {
+    console.log(`Notifying beneficiary with userId: ${userId}`)
+
+    // Create a nodemailer transporter
+    const transporter = nodemailer.createTransport({
+      service: 'gmail',
+      auth: {
+        user: process.env.GMAIL_USER,
+        pass: process.env.GMAIL_PASSWORD,
+      },
+    })
+
+    // Compose the email content
+    const mailOptions = {
+      from: process.env.GMAIL_USER,
+      to: user.email,
+      subject: 'Cash Out Ready',
+      text: `Hi, ${user.firstName} ${user.lastName} \n\nCongratuations! Your cashout is ready \n\n  Please navigate to your account and cashout your funds from your group and get ready for the next contribution if your group will still be active. Thank you and enjoy your funds.`,
+    }
+
+    // Send the email
+    transporter.sendMail(
+      mailOptions,
+      (err: Error | null, info: { response: string }) => {
+        if (err) {
+          console.log(
+            `Failed to send cashout email. Please try again later.`,
+            err
+          )
+        } else {
+          console.log(
+            'Cashout email has been sent to your email if you have an account with us: ' +
+              info.response
+          )
+        }
+      }
+    )
+  } else {
+    console.log('No next beneficiary to notify.')
+  }
```

Comment:
I think we have an emailFactory function that handles this

Reply:
https://github.com/Ajo-Savings-Project/ajo-savings/blob/main/apps/server/src/utils/createEmailFactory.ts

Reply:
use as 

```
const {sendMail} = createEmail({
  to,
  subject,
  templatePath,
  templateContext,
})

await sendMail()
```

Reply:
Okay boss